api/client/import.go

```diff
@@ -52,6 +54,15 @@ func (cli *DockerCli) CmdImport(args ...string) error {
 
 	if src == "-" {
 		in = cli.in
+	} else if !strings.HasPrefix(src, "http://") && !strings.HasPrefix(src, "https://") {
+		v.Set("fromSrc", "-")
+		file, err := os.Open(src)
+		defer file.Close()
+		in = file
+		if err != nil {
```

Comment:
Just for readability purpose, I'd rather see that test right after the call to `os.Open` (and we don't even have to defer `file.Close()` if `err != nil` anyways).